### PR TITLE
[Runtime] Allow JSON for APP_RUNTIME_OPTIONS

### PR DESCRIPTION
**Pull request by ro0NL**
| Q             | A
| ------------- | ---
| Branch?       | 6.1
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

So we can export `APP_RUNTIME_OPTIONS='{"disable_dotenv":true}'` in prod

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
This is not a bugfix ;)
~What about making SymfonyRuntime's constructor accept array|json-string instead?~

File: src/Symfony/Component/Runtime/Internal/autoload_runtime.template [hunk 1]
```diff
@@ -18,8 +18,13 @@ if (!is_object($app)) {
     throw new TypeError(sprintf('Invalid return value: callable object expected, "%s" returned from "%s".', get_debug_type($app), $_SERVER['SCRIPT_FILENAME']));
 }
 
+$options = $_SERVER['APP_RUNTIME_OPTIONS'] ?? $_ENV['APP_RUNTIME_OPTIONS'] ?? [];
+if (!is_array($options) && is_array($json = json_decode($options, true))) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Runtime/Internal/autoload_runtime.template]:
```php
if (is_string($options = $_SERVER['APP_RUNTIME_OPTIONS'] ?? $_ENV['APP_RUNTIME_OPTIONS'] ?? [])) {
    $options = json_decode($options, true, JSON_THROW_ON_ERROR);
}
```

File: src/Symfony/Component/Runtime/Internal/autoload_runtime.template [hunk 2]
```diff
@@ -18,8 +18,13 @@ if (!is_object($app)) {
...
+    $options = $json;
+}
+
 $runtime = $_SERVER['APP_RUNTIME'] ?? $_ENV['APP_RUNTIME'] ?? %runtime_class%;
-$runtime = new $runtime(($_SERVER['APP_RUNTIME_OPTIONS'] ?? $_ENV['APP_RUNTIME_OPTIONS'] ?? []) + %runtime_options%);
+$runtime = new $runtime($options + %runtime_options%);
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Runtime/Internal/autoload_runtime.template]: `$options += %runtime_options%` for better global state reflection

**Review by nicolas-grekas — COMMENTED** [review]
>  What about making SymfonyRuntime's constructor accept array|json-string instead?

Not a good idea as that would prevent merging with options generated by the plugin!

**Comment by ro0NL** [comment]
> This is not a bugfix ;)

Are you saying APP_RUNTIME_OPTIONS is not a supported env var? but more like some internal global var?

IMHO it looks like an env var extension point ... thus should support string values actually. Thus a bugfix :)

Or how could i set an PHP array using env var?

We're not at 6.1 yet, so as a feature i will postpone this PR. And we patch the front controller manually for the time being (maybe forever), if that's the intended feature-usage currently.

**Comment by ro0NL** [comment]
So, tbe bugfix would be

```
$ APP_RUNTIME_OPTIONS={} bin/console 
PHP Warning:  A non-numeric value encountered in vendor/autoload_runtime.php on line 23
PHP Fatal error:  Uncaught Error: Unsupported operand types in vendor/autoload_runtime.php:23
```

**Comment by nicolas-grekas** [comment]
> more like some internal global var?

yes, that's been the intent so far, not reading from env.

**Comment by ro0NL** [comment]
then, what about renaming it to `SF_RUNTIME_OPTIONS` for 6.1?

**Comment by nicolas-grekas** [comment]
I don't see why, better not to me :shrug:

**Comment by nicolas-grekas** [comment]
Can you add a test case?

**Comment by ro0NL** [comment]
I'll continue this with our php8/sf6 upgrade :+1:

File: src/Symfony/Component/Runtime/Internal/autoload_runtime.template [hunk 3]
```diff
@@ -18,8 +18,13 @@ if (!is_object($app)) {
     throw new TypeError(sprintf('Invalid return value: callable object expected, "%s" returned from "%s".', get_debug_type($app), $_SERVER['SCRIPT_FILENAME']));
 }
 
+if (is_string($options = $_SERVER['APP_RUNTIME_OPTIONS'] ?? $_ENV['APP_RUNTIME_OPTIONS'] ?? [])) {
+    $options = json_decode($options, true, JSON_THROW_ON_ERROR);
```
Comment (ro0NL) [inline src/Symfony/Component/Runtime/Internal/autoload_runtime.template:22]: for future referene, there's no guarantee we decode an array

**Pull request closed without merging**